forge-game/src/main/java/forge/game/ability/effects/EffectEffect.java

```diff
@@ -134,15 +134,24 @@ public void resolve(SpellAbility sa) {
             name = hostCard + (sa.hasParam("Boon") ? "'s Boon" : "'s Effect");
         }
 
-        // Unique Effects shouldn't be duplicated
-        if (sa.hasParam("Unique") && game.isCardInCommand(name)) {
-            return;
+        PlayerCollection effectOwner = sa.hasParam("EffectOwner") ?
+                AbilityUtils.getDefinedPlayers(hostCard, sa.getParam("EffectOwner"), sa) :
+                new PlayerCollection();
+        if (effectOwner.isEmpty()) {
```

Comment:
If param isn't used you need to initialize it directly with `getActivatingPlayer()`
Otherwise you'd have changed the logic slighty (if Defined doesn't yield results it would add undesired owner)